### PR TITLE
Chat coc

### PR DESCRIPTION
**Pull request by dudethatbe**
added input and CoC agree terms for non-registered chat users

### CODE REVIEW

File: app/templates/components/datafruits-chat.hbs [hunk 1]
```diff
@@ -34,33 +34,52 @@
...
+            class="cool-button mt-1 md:m-0" {{action "toggleGifsEnabled"}}>
+            {{t (if this.gifsEnabled  "chat.img_on" "chat.img_off") }}
+          </button>
+        </div>
+        <div>
+          <Input
+            name="agree-to-coc"
+            id="agree-to-coc"
+            @type="checkbox"
+            @checked={{agreeToCoC}}
```
Comment (mcfiredrill) [inline app/templates/components/datafruits-chat.hbs:66]:
```suggestion
            @checked={{this.agreeToCoC}}
```

File: app/templates/components/datafruits-chat.hbs [hunk 2]
```diff
@@ -34,33 +34,52 @@
...
+            {{t (if this.gifsEnabled  "chat.img_on" "chat.img_off") }}
+          </button>
+        </div>
+        <div>
+          <Input
+            name="agree-to-coc"
+            id="agree-to-coc"
+            @type="checkbox"
+            @checked={{this.agreeToCoC}}
+            aria="Agree with the Code of Conduct to join the chat"
```
Comment (mcfiredrill) [inline app/templates/components/datafruits-chat.hbs:67]:
```suggestion
            aria={{t "coc.checkbox_aria"}}
```
or something

File: translations/es.json [hunk 3]
```diff
@@ -22,7 +22,9 @@
       "nick": "apodo",
       "input": "mensaje",
       "search": "buscar por etiqueta",
-      "aria": "Visita la sala de chat de datafruits.fm"
+      "aria": "Visita la sala de chat de datafruits.fm",
+      "coc": "Code of Contact",
+      "i_agree": "He leído el Código de Conducta y acepto seguir sus términos y entiendo que pueden ser removidos por no hacerlo."
```
Comment (mcfiredrill) [inline translations/es.json:27]: I couldn't find where this key was being used.

Comment (mcfiredrill) [reply translations/es.json:27]: And where is the English version ?

**Comment by mcfiredrill** [comment]
Functionally seems good, gave it a quick test locally. :100: 

I know we have been a bit lax on writing tests...perhaps we could add one to test the behavior of the button being disabled or not in this integration test:
https://github.com/datafruits/datafruits/blob/master/tests/integration/components/datafruits-chat-test.js

**Comment by bradwbradw** [comment]
https://datafruits-r-chat-coc-6ogscrlz.herokuapp.com/ is showing an application error screen (commenting just in case you weren't aware)

**Comment by mcfiredrill** [comment]
> https://datafruits-r-chat-coc-6ogscrlz.herokuapp.com/ is showing an application error screen (commenting just in case you weren't aware)

Unfortunately yeah I haven't got review apps quite working, but I'm making progress, soon hopefully. :v:

**Comment by mcfiredrill** [comment]
@dudethatbe If you merge master and push to this branch, I should be able to bring up a review app for this PR.

File: translations/es.json (hunk 3 above)
Comment (dudethatbe) [reply translations/es.json:27]: Thanks for catching these, they aren't needed so I'll remove them

File: app/templates/components/datafruits-chat.hbs (hunk 2 above)
Comment (dudethatbe) [reply app/templates/components/datafruits-chat.hbs]: Ah woops, yeah that should be a translation, not  a plain string

**Comment by dudethatbe** [comment]
> I know we have been a bit lax on writing tests...perhaps we could add one to test the behavior of the button being disabled or not in this integration test:
> https://github.com/datafruits/datafruits/blob/master/tests/integration/components/datafruits-chat-test.js

I'm having troubles figuring out how to select the "Join Chat" button from the template in such a way that i can test its disabled property.
https://github.com/datafruits/datafruits/pull/584/commits/980eb2ec0393b2f8c1618c8e99a79bd6652418e2
I tried logging the `join` variable to see if it's grabbing it or not, but I wasn't able to see anything in the console to help me out with that. Anyways, it seems like its almost there, just need some help with selecting the right element :sweat_smile:

File: app/components/datafruits-chat.js [hunk 4]
```diff
@@ -30,8 +30,13 @@ export default class DatafruitsChat extends Component {
   joinedUsers;
 
   @computed('isJoiningChat', 'nick.length')
+  @action
+  agreeChanged() {
+    // input handler requires a function to exist for this to work correctly
+    // ¯\_(~_~)_/¯
+  }
```
Comment (mcfiredrill) [inline app/components/datafruits-chat.js:37]: It works for me if I remove this and add `agreeToCoC` to the list of computed dependencies above.

**Comment by mcfiredrill** [comment]
I've got a review app up here:
https://datafruits-review-pr-597.herokuapp.com/chat

**Comment by dudethatbe** [comment]
After testing it out, I wanted to see if I could join the chat without agreeing with the CoC, and it turns out that removing the `disabled` property from the button to join the chat is all it takes. As flimsy as this is, I don't know if it's worth the time to further improve it. It's just a website right?

**Comment by mcfiredrill** [comment]
> After testing it out, I wanted to see if I could join the chat without agreeing with the CoC, and it turns out that removing the disabled property from the button to join the chat is all it takes. As flimsy as this is, I don't know if it's worth the time to further improve it. It's just a website right?

If someone is messing around with the HTML then clearly they wouldn't care about any terms we have for people to agree with anyway IMO.

**Comment by mcfiredrill** [comment]
Merged in https://github.com/datafruits/datafruits/pull/597

**Pull request merged**